fix: 429 when querying wiki images

oscargolding opened the pull request:
Hey folks,

Recently (in the past month or so on v2025.10) I noticed that on large wiki articles none of the images are appearing. After enabling the logs, I saw the following error suggesting that rate limiting has now started happening from wiki:

```txt
12/28/25-17:16:18 WARN  HTTP status not okay: HTTP/1.1 429 Use thumbnail steps listed on https://w.wiki/GHai. Please contact noc@wikimedia.org for further information (a765913)
12/28/25-17:16:18 DEBUG Response headers: {
  ["access-control-allow-origin"] = "*",
  ["access-control-expose-headers"] = "Age, Date, Content-Length, Content-Range, X-Content-Duration, X-Cache",
  connection = "close",
  ["content-length"] = "2147",
  ["content-type"] = "text/html; charset=utf-8",
  date = "Sun, 28 Dec 2025 06:16:18 GMT",
  nel = "{ \"report_to\": \"wm_nel\", \"max_age\": 604800, \"failure_fraction\": 0.05, \"success_fraction\": 0.0}",
  ["report-to"] = "{ \"group\": \"wm_nel\", \"max_age\": 604800, \"endpoints\": [{ \"url\": \"https://intake-logging.wikimedia.org/v1/events?stream=w3c.reportingapi.network_error&schema_uri=/w3c/reportingapi/network_error/1.0.0\" }] }",
  ["retry-after"] = "10",
  server = "Varnish",
  ["server-timing"] = "cache;desc=\"int-front\", host;desc=\"cp5028\"",
  ["strict-transport-security"] = "max-age=106384710; includeSubDomains; preload",
  ["timing-allow-origin"] = "*",
  ["x-analytics"] = "",
  ["x-cache"] = "cp5028 int",
  ["x-cache-status"] = "int-front",
  ["x-client-ip"] = "<blah>",
  ["x-request-id"] = "e42df733-22c9-4914-a419-0f07b55d5470"
} --[[table: 0xb3a3dc00]]
12/28/25-17:16:18 WARN  failed fetching image from https://upload.wikimedia.org/wikipedia/commons/thumb/f/fa/Flag_of_the_People%27s_Republic_of_China.svg/211px-Flag_of_the_People%27s_Republic_of_China.svg.png : Remote server error or unavailable
```

If you look into the suggested link https://w.wiki/GHai it links to this recent change on their end https://gerrit.wikimedia.org/r/c/mediawiki/core/+/1121793 - koreader just needs to become compliant. 

Confirmed that on my kobo libra color that with this fix images go from not appearing to appearing now. 

Apologies if I haven't done something correctly - this is a first time contribution. I did make https://github.com/koreader/koreader/pull/14763 but closed it since my theory for the error was not correct (at the time).

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/koreader/koreader/14765)
<!-- Reviewable:end -->

Frenzie commented:
You can simply update the PR with new changes. At a glance the changes in this PR don't consider the smaller size vs higher quality choice?

oscargolding commented:
Apologies didn't pull down koreader and I deleted that branch, moved to https://github.com/koreader/koreader/pull/14766 which has the right setup now. 

> At a glance the changes in this PR don't consider the smaller size vs higher quality choice?

Both the high quality and low quality image use the domain prefix `https://upload.wikimedia.org/wikipedia/commons/thumb` which is now limited to the values in the documentation linked, hence why both must fork out to the step choice function.
